src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -370,53 +463,81 @@ class npc_ichor_globule : public CreatureScript
             return;
         }
 
-        void UpdateAI(uint32 uiDiff) override
+        void SetData(uint32 id, uint32 data) override
         {
-            if (uiRangeCheck_Timer < uiDiff)
+            if (id == 0)
             {
-                if (Creature* pIchoron = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_ICHORON)))
-                {
-                    if (me->IsWithinDist(pIchoron, 2.0f, false))
-                    {
-                        if (pIchoron->AI())
-                            pIchoron->AI()->DoAction(ACTION_WATER_ELEMENT_HIT);
-                        me->DespawnOrUnsummon();
-                    }
-                }
-                uiRangeCheck_Timer = 1000;
+                pathId = data;
+                me->GetMotionMaster()->MovePoint(0, globulePaths[pathId]);
             }
-            else uiRangeCheck_Timer -= uiDiff;
         }
 
-        void JustDied(Unit* /*killer*/) override
+        void MovementInform(uint32 type, uint32 id) override
+        {
+            if (type != POINT_MOTION_TYPE)
+                return;
+
+            switch (id)
+            {
+            case 0:
+                me->GetMotionMaster()->MovementExpired();
+                events.ScheduleEvent(1, 500);
+                break;
+            case 1:
+                me->GetMotionMaster()->MovementExpired();
+                if (Creature* ichoron = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_ICHORON)))
+                    me->GetMotionMaster()->MoveFollow(ichoron, 0.0f, 0.0f);
+                break;
+            }
+        }
+
+        // on retail spell casted on a creature's death are not casted after death but keeping mob at 1 health, casting it and then letting the mob die.
+        // splash is implemented in this way
+        void DamageTaken(Unit* attacker, uint32 &damage) override
         {
-            DoCast(me, SPELL_SPLASH);
-            if (Creature* pIchoron = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_ICHORON)))
-                if (pIchoron->AI())
-                    pIchoron->AI()->DoAction(ACTION_WATER_ELEMENT_KILLED);
+            int32 actualHp = (int32)me->GetMaxHealth() * (me->GetHealthPct() / 100);
```

Comment:
https://ideone.com/85wyY1
I get that this is a work in progress and it's untested and all that but it should be clear to the developer mind that's the behavior of / operator with integers and what's the order of precedence http://www.cppreference.com/wiki/operator_precedence.

This whole PR should be debugged line by line once finished to ensure the "rework" actually adds new working functionalities